docs/reference/manifest.rst

```diff
@@ -1,35 +1,176 @@
+.. _manifest:
+
 MicroPython manifest files
 ==========================
 
-When building firmware for a device the following components are included in
-the compilation process:
+Summary
+-------
+
+MicroPython has a feature that allows Python code to be "frozen" into the
+firmware, as an alternative to loading code from the filesystem.
+
+This has the following benefits:
+
+- the code is pre-compiled to bytecode, avoiding the need for the Python
+  source to be compiled at load-time.
+- the bytecode can be executed directly from ROM (i.e. flash memory) rather than
+  being copied into RAM. Similarly any constant objects (strings, tuples, etc)
+  are loaded from ROM also. This can lead to significantly more memory being
+  available for your application.
+- on devices that do not have a filesystem, this is the only way to
+  load Python code.
+
+During development, freezing is generally not recommended as it will
+significantly slow down your development cycle, as each update will require
+re-flashing the entire firmware. However, it can still be useful to
+selectively freeze some rarely-changing dependencies (such as third-party
+libraries).
+
+The way to list the Python files to be be frozen into the firmware is via
+a "manifest", which is a Python file that will be interpreted by the build
+process. Typically you would write a manifest file as part of a board
+definition, but you can also write a stand-alone manifest file and use it with
+an existing board definition.
+
+Manifest files can define dependencies on libraries from :term:`micropython-lib`
+as well as Python files on the filesystem, and also on other manifest files.
+
+Writing manifest files
+----------------------
+
+A manifest file is a Python file containing a series of function calls. See the
+available functions defined below.
+
+Any paths used in manifest files can include the following variables. These all
+resolve to absolute paths.
+
+- ``$(MPY_DIR)`` -- path to the micropython repo.
+- ``$(MPY_LIB_DIR)`` -- path to the micropython-lib submodule. Prefer to use
+  ``require()``.
+- ``$(PORT_DIR)`` -- path to the current port (e.g. ``ports/stm32``)
+- ``$(BOARD_DIR)`` -- path to the current board
+  (e.g. ``ports/stm32/boards/PYBV11``)
+
+Custom manifest files should not live in the main MicroPython repository. You
+should keep them in version control with the rest of your project.
+
+Typically a manifest used for compiling firmware will need to include the port
+manifest, which might include frozen modules that are required for the board to
+function. If you just want to add additional modules to an existing board, then
+include the board manifest (which will in turn include the port manifest).
+
+The ``opt`` kwarg can be set on the various functions, this controls the
```

Comment:
kwarg -> keyword argument

also, maybe this paragraph makes more sense down below, closer to the functions themselves?